js/admin.block-post.js

```diff
@@ -54,6 +63,64 @@
 			$( this ).hide();
 		});
 
+		$( '.block-properties-category-create-button' ).on( 'click', function( e ) {
+			e.preventDefault();
+			$( this ).hide();
+			$( '#block-properties-category-create-wrapper' ).show().find( 'input[type="text"]:first').focus();
+		});
+
+		$( '.block-properties-category-remove-button' ).on( 'click', function( e ) {
+			e.preventDefault();
+			$( this ).parent().hide();
+			$( '.block-properties-category-create-button' ).show();
+		});
+
+		$( '.block-properties-category-remove-button' ).on( 'click', function( e ) {
+			e.preventDefault();
+			$( this ).parent().hide();
+			$( '.block-properties-category-create-button' ).show();
+		});
+
+		$( '.block-properties-category-save-button' ).on( 'click', function( e ) {
+			e.preventDefault();
+			let categoryVal = $.trim( $( '#block-properties-category-name' ).val() );
+			if ( '' === categoryVal ) {
+
+				// Show error message.
+				$( '#block-properties-category-create-status' ).removeClass().addClass( 'error' ).html( blockLab.emptyCategory );
+				$( '#block-properties-category-create-wrapper' ).find( 'input[type="text"]:first').focus();
+				return;
+			}
+			$( this ).html( blockLab.saving ).attr( 'disabled', 'disabled' );
+			wp.ajax.send( 'save_custom_category', {
```

Comment:
If we just save the category as post meta, and register it when the blocks are being registered, then there's no need to actually save the category independently at any point – and therefore no need for this ajax call.

Reply:
My main concern is category re-use across the site and in a multisite setup. If we store categories as an independent option, there's re-use rather than storing as post meta for a per-site setup. What are your thoughts?

Reply:
@lukecarbis I've posted a new video with the changes and my justification to keep Ajax saving and a site-wide option for categories for MS installs.

https://www.dropbox.com/s/r8urivzxmckp1uk/block-lab-custom-categories.mp4?dl=0

Reply:
@ronaldhuereca Given that blocks themselves don't work across sites, I don't think it's worth introducing a new site wide option to keep categories working across sites. I think it's reasonable to expect that categories won't follow you between sites. It might even be preferable.

I don't actually think it should be post meta, per se, but just saved in exactly the same way as the current category gets saved (which is inside a JSON blob).